hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMBlockProtocolServer.java

```diff
@@ -196,6 +196,16 @@ public List<AllocatedBlock> allocateBlock(
           blocks.add(block);
```

Comment:
Thanks @aswinshakil for the question.  I don't know why `ScmBlockLocationProtocolServerSideTranslatorPB` throws the exception instead of `SCMBlockProtocolServer` for "not enough blocks".  It seems to be unique to this method.  CC @mukul1987 @sodonnel

Reply:
Yeah that is BlockManagerImpl.allocateBlock throws an exception when allocate block fails in the selected container. But if the container is null, a null is returned. Changes need to go in  SCMBlockProtocolServer to fail it.